### PR TITLE
PYSCAN-50 Document the setup of poetry to pull dependencies from private artifactory repo

### PR DESCRIPTION
**Pull request by jdodinh**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -83,3 +83,6 @@ target-version = ["py38", "py39", "py310", "py311", "py312"]
 [tool.coverage.run]
 branch = true
 relative_files = true
+
+[tool.pytype]
+inputs = ['src']
```
Comment (joke1196) [inline pyproject.toml:88]: I don't think we use pytype in the project. The dependency would be missing. If you want to use it, I think this would belong in a different PR/ticket.

File: README.md [hunk 2]
```diff
@@ -139,7 +139,15 @@ Run `python src/py_sonar_scanner`
...
-Run `poetry install` to install the dependencies
+Run `poetry install` to install the dependencies. By default, the dependencies are installed from a private repository. 
+
+If you wish to install the dependencies from the public PyPI repository, remove the following from `pyproject.toml`:
+```toml
+[[tool.poetry.source]]
+name = "repox"
+url = "https://repox.jfrog.io/artifactory/api/pypi/sonarsource-pypi/simple/"
+priority = "default"
+```
```
Comment (joke1196) [inline README.md:150]: I think this is quite nice, but I guess the main point here would be to explain how to setup the authentication to our repox for poetry, this is [explained here](https://python-poetry.org/docs/repositories/) and maybe the local setup should be explained by setting:
```
export POETRY_HTTP_BASIC_PYPI_USERNAME=<username>
export POETRY_HTTP_BASIC_PYPI_PASSWORD=<password> 
```
I guess just a quick sentence as well explaining the PASSWORD should be the api token created on your account in repox.

**Review by joke1196 — CHANGES_REQUESTED** [review]
Looks good! I would just go into the detail to setup the auth locally. So other devs would be able to pull the correct  dependencies.

File: pyproject.toml (hunk 1 above)
Comment (jdodinh) [reply pyproject.toml:88]: ooops, this was not meant to be in this PR sorry...

**Review by joke1196 — APPROVED** [review]
LGTM! Thank you!
